task/sensor-api/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Client API for the Sensor task.
+
+#![no_std]
+
+use drv_i2c_api::ResponseCode;
+use userlib::*;
+
+#[derive(zerocopy::AsBytes, Copy, Clone, Debug, PartialEq)]
+#[repr(C)]
+pub struct SensorId(pub usize);
+
+impl From<usize> for SensorId {
+    fn from(id: usize) -> Self {
+        SensorId(id)
+    }
+}
+
+impl From<SensorId> for usize {
+    fn from(id: SensorId) -> Self {
+        id.0
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+pub enum Reading {
+    None,
```

Comment:
Consider eliminating `None` and putting this in an option where it might be missing, so that people who know it _isn't_ missing don't have to cope with a non-removable `None` variant.

Reply:
Yes, good call.